src/colmap/sensor/models.h

```diff
@@ -127,8 +127,12 @@ enum class CameraModelId {
       double focal_length, size_t width, size_t height);                      \
   template <typename T>                                                       \
   static void ImgFromCam(const T* params, T u, T v, T w, T* x, T* y);         \
-  template <typename T>                                                       \
-  static void CamFromImg(const T* params, T x, T y, T* u, T* v, T* w);        \
+  static inline void CamFromImg(const double* params,                         \
```

Comment:
Rather than inlining these, would it make sense to move the implementation to the .cc file now? Same goes for the IterativeUndistortion.

Reply:
May I ask what s the benefit of moving ``CamFromImg`` to cc? This will require moving the implementation of ``CamFromImg`` for all classes to cc. Does it make the code structure cleaner? If so I could do it. 

I can see a clear benefit of moving ``IterativeUndistortion`` to cc for faster development, but we unfortunately can not do it in an elegant way as it is a method of a template class with CameraModel as its template.

Reply:
I think you already mentioned one reason: fewer recompilations upon code changes in the CamFromImg() methods. Otherwise, fewer inlined code thus faster compilation and possibly smaller overall binary sizes. Unless we have some profiling that shows us that we really want to inline the method on all the callsites, I suggest to move it to .cc. I can do this in a follow-up PR though, as it is not directly related to the changes here.

Reply:
I think ``CamFromImg`` will very rarely be touched upon in its current state. But I agree that it can benefit performance. I will leave it for the follow-up PR. Thanks : )

Reply:
Thanks, I'll follow up.